### PR TITLE
[distillation] Add reverse KL student top-k loss

### PR DESCRIPTION
**Pull request by YichenZW**
## What

Add `reverse_kl_student_topk` as a top-k distillation loss mode.

This includes:
- FSDP/VeOmni reverse KL student top-k support
- Megatron reverse KL student top-k support
- teacher log-prob lookup on student top-k ids
- raw mass metrics before clamp
- missing teacher fallback excluded from teacher mass
- CPU regression coverage for metrics and Megatron dispatch

## Tests

- `pytest -q tests/workers/test_distillation_topk_symmetry_on_cpu.py`
- Megatron world-size=1 mock forward/backward parity against a pure PyTorch reference

### CODE REVIEW

File: verl/trainer/distillation/megatron/losses.py [hunk 1]
```diff
@@ -261,6 +282,123 @@ def backward(
         return grad_input, None, None, None
 
 
+class _VocabParallelReverseKLDivergenceOnStudentTopK(torch.autograd.Function):
```
Comment (Luosuu) [inline verl/trainer/distillation/megatron/losses.py:285]: maybe next PR for megatron :)

File: verl/trainer/distillation/fsdp/losses.py [hunk 2]
```diff
@@ -72,6 +72,41 @@ def kl_divergence(log_q: torch.Tensor, log_p: torch.Tensor) -> torch.Tensor:
...
+def reverse_kl_divergence(log_q: torch.Tensor, log_p: torch.Tensor) -> torch.Tensor:
+    """Compute KL(q || p) from log probabilities on a truncated support."""
+    log_p = log_p.float()
+    log_q = log_q.float()
+    q = log_q.exp()
+    kld = q * (log_q - log_p)
+    return kld.sum(dim=-1)
+
+
+def _gather_teacher_log_probs_on_student_topk(
```
Comment (Luosuu) [inline verl/trainer/distillation/fsdp/losses.py:84]: maybe do this in another PR. this version can just add a naive implementation that gets the full teacher logits to sample student topk logprobs

File: verl/trainer/distillation/fsdp/losses.py [hunk 3]
```diff
@@ -147,3 +182,82 @@ def compute_forward_kl_topk(
...
+    if student_topk is None:
+        raise ValueError("reverse_kl_student_topk requires distillation_loss.topk.")
+    teacher_logprob_topk = teacher_topk_ids.shape[-1]
+    if teacher_logprob_topk < student_topk:
+        raise ValueError(
+            "Teacher log-prob payload for reverse_kl_student_topk must be at least as wide as "
+            f"distillation_loss.topk, but got {teacher_logprob_topk} < {student_topk}."
+        )
+    use_chunked_topk = getattr(loss_config, "use_chunked_topk", False)
+    if use_chunked_topk:
```
Comment (Luosuu) [inline verl/trainer/distillation/fsdp/losses.py:223]: we can make this always true

File: verl/trainer/distillation/fsdp/losses.py [hunk 4]
```diff
@@ -147,3 +182,82 @@ def compute_forward_kl_topk(
         "overlap_count": overlap_count,
         "overlap_token_advantage": overlap_token_advantage,
     }
+
+
+def compute_reverse_kl_student_topk(
+    student_logits: torch.Tensor,
```
Comment (Luosuu) [inline verl/trainer/distillation/fsdp/losses.py:188]: is this full vocab logits?
[num_tok, vocab]?

**Review by Luosuu — COMMENTED** [review]
overall a very good feature to add. please simplify a bit for the first PR
